uktrade/data-hub-frontend
# Add linting rule for separating imports

## Description of change

An issue that has come up recently in PR reviews is that third party imports aren't being separated from local ones. There are a few reasons for this, ranging from human error to inconsistencies within the codebase.

In order to remedy this, I have added a new linting rule that will catch these import issues before PRs are raised. This uses the ESLint [import/order rule](https://github.com/import-js/eslint-plugin-import/blob/v2.28.1/docs/rules/order.md) and will raise linting errors when incorrectly grouped imports are detected, which can then be fixed like any other linting error.

I have applied this new rule across the codebase. There are no changes to these files apart from making the fixes suggested by the linter (with a couple of exceptions relating to cleaning up duplicate imports I identified).

## Test instructions
Attempting to add a third party and local import without a newline inbetween should now raise an error:
<img width="679" alt="Screenshot 2023-09-27 at 16 57 08" src="https://github.com/uktrade/data-hub-frontend/assets/36161814/e97868b8-d846-4333-ad28-246822f1b292">

Attempting to mix third party and local imports should now raise an error:
<img width="751" alt="Screenshot 2023-09-27 at 16 57 37" src="https://github.com/uktrade/data-hub-frontend/assets/36161814/9d619038-aab1-41a8-84f4-fe4047937044">

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
